common/pulp_docker/common/tarutils.py

```diff
@@ -114,3 +114,24 @@ def get_youngest_children(metadata):
                 pass
 
     return list(image_ids)
+
+
+def get_image_manifest(tarfile_path):
+    """
+    Given a path to a tarfile, which itself is the product of "skopeo copy"
+    this returns the decoded manifest.json file if it exists.
+
+    :param tarfile_path: full path to tarfile
+    :type tarfile_path:  basestring
+    :return:             decoded manifest json
+    :rtype               list or dict
+    """
+    image_manifest = []
```

Comment:
What about setting this to None, so the here https://github.com/pulp/pulp_docker/pull/203/files#diff-2d828905f567587f4d447f29972967f8R38 you can just check, in case it is None then we know that the tar uploaded is a tar or V1 image.